packages/api/prisma/schema.prisma

```diff
@@ -32,81 +50,106 @@ type ArticleRevision {
   socialMediaAuthorIDs   String[]
   socialMediaImageID     String?
 
-  revision    Int
+  revision    Int       @default(0)
   createdAt   DateTime  @default(now())
+  modifiedAt  DateTime? @updatedAt
   updatedAt   DateTime?
   publishAt   DateTime?
   publishedAt DateTime?
+
+  PublishedArticle Article[] @relation("publishedArticleRevision")
+  PendingArticle   Article[] @relation("pendingArticleRevision")
+  DraftArticle     Article[] @relation("draftArticleRevision")
+
+  @@index([publishAt])
+  @@index([publishedAt])
+  @@index([updatedAt])
+  @@index([tags])
+  @@map("articles.revisions")
 }
 
 model Article {
-  id         String           @id @default(cuid()) @map("_id")
-  createdAt  DateTime         @default(now())
-  modifiedAt DateTime         @updatedAt
-  published  ArticleRevision?
-  pending    ArticleRevision?
-  draft      ArticleRevision?
-  shared     Boolean
-
-  @@index([createdAt], map: "createdAt_-1")
-  @@index([modifiedAt], map: "modifiedAt_-1")
-  @@index([published.publishedAt], map: "published.publishedAt_-1")
-  @@index([published.updatedAt], map: "published.updatedAt_-1")
-  @@index([pending.publishAt], map: "pending.publishAt_-1")
-  @@index([draft.tags], map: "draft.tags_1")
-  @@index([pending.tags], map: "pending.tags_1")
-  @@index([published.tags], map: "published.tags_1")
-  @@map("articles")
-}
+  id         String   @id @default(cuid())
+  createdAt  DateTime @default(now())
+  modifiedAt DateTime @updatedAt
+
+  // The better way in the future to go around draft, pending, published is to have the articleId and a state on the revision
+  // There should then be a unique constraint on id + state.
+  // This would allow us to
+  //  cascade the revisions on article delete
```

Comment:
Wdyt about keeping revisions forever ? And indeed deletion would just delete the wrapper object?

Reply:
I totally agree with this, will need a few changes before we can do that (the way publishing, updating, etc. works)